promql/engine_test.go

```diff
@@ -739,8 +786,38 @@ load 10s
 			Interval: 5 * time.Second,
 		},
 		{
-			Query:      "rate(bigmetric[1s])",
-			MaxSamples: 1,
+			Query:        `sum(metric)`,
+			MaxSamples:   8,
+			TotalSamples: 9,
```

Comment:
This seems too high. It reads in 3 samples, and outputs 3 - so I'd expect 6.

Reply:
Bringing the Increment out from the inner loop (previous comments) in rangeEval, where we are checking the `ts` comparison is the culprit. We need to make sure to consider only those that match the timestamp else we are considering all the series. Thats the reason why we are getting 9 and not 6. Fixing this back to the inner loop.

Reply:
No, the issue was that we were using the wrong variable. vectors[i] is what we should have been taking the length of.